storage/remote/queue_manager.go

```diff
@@ -692,19 +742,18 @@ func (t *QueueManager) calculateDesiredShards() int {
 	// will need to do next iteration.  We add to this any pending samples
 	// (received - send) so we can catch up with any backlog. We use the average
 	// outgoing batch latency to work out how many shards we need.
-	var (
-		samplesInRate      = t.samplesIn.rate()
-		samplesOutRate     = t.samplesOut.rate()
-		samplesKeptRatio   = samplesOutRate / (t.samplesDropped.rate() + samplesOutRate)
-		samplesOutDuration = t.samplesOutDuration.rate() / float64(time.Second)
-		samplesPendingRate = samplesInRate*samplesKeptRatio - samplesOutRate
-		highestSent        = t.metrics.highestSentTimestamp.Get()
-		highestRecv        = t.highestRecvTimestamp.Get()
-		delay              = highestRecv - highestSent
-		samplesPending     = delay * samplesInRate * samplesKeptRatio
-	)
-
-	if samplesOutRate <= 0 {
+	var c ShardingCalculations
```

Comment:
Can't tell if this version using the struct is too much uglier, and if it would make more sense to just copy the values in at the end?

Reply:
I slightly prefer copying the stats at the end, but not a big deal.